router/src/main/java/io/micronaut/web/router/DefaultRouter.java

```diff
@@ -75,11 +75,12 @@ public class DefaultRouter implements Router, HttpServerFilterResolver<RouteMatc
     private Set<Integer> ports;
     private final List<FilterRoute> alwaysMatchesFilterRoutes = new ArrayList<>();
     private final List<FilterRoute> preconditionFilterRoutes = new ArrayList<>();
-    private final Supplier<List<GenericHttpFilter>> alwaysMatchesHttpFilters = SupplierUtil.memoized(() -> {
+    // ArrayList to avoid interface checkcast
+    private final Supplier<ArrayList<GenericHttpFilter>> alwaysMatchesHttpFilters = SupplierUtil.memoized(() -> {
         if (alwaysMatchesFilterRoutes.isEmpty()) {
-            return Collections.emptyList();
+            return new ArrayList<>(0);
```

Comment:
Please add a comment why this cannot be emptyList or List.of

Reply:
I changed the field to be ArrayList to avoid an interface cast, so this lambda has to return ArrayList as well